pkg/virt-controller/watch/migration.go

```diff
@@ -1037,6 +1055,24 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 		return fmt.Errorf("vmi is inelgible for migration because another migration job is running")
 	}
 
+	if migration.DeletionTimestamp != nil && !migration.IsFinal() {
+		err = c.handleSignalMigrationAbort(migration, vmi)
+		if err != nil {
+			return err
+		}
+
+		if migration.Status.Phase != virtv1.MigrationRunning && targetPodExists && pod.Status.Phase == k8sv1.PodPending {
+			// In this case when the migration has been aborted and target pod is still not ready and migration did
+			// not start to run, delete target pod
+			err = c.clientset.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, v1.DeleteOptions{})
+			if err != nil {
+				return fmt.Errorf("cannot delete pending target pod %s/%s for migration although migration is aborted", pod.Name, pod.Namespace)
+			}
+
+			return nil
+		}
+	}
```

Comment:
Should we return here or continue normal execution?

Reply:
I don't think so. The reason is to allow `handlePendingPodTimeout()` to be called if necessary.

For example, consider a case where the target pod is "Scheduling" and for some reason it hangs and cannot actually start. In this case we want to kill and clean it after timeout exceeds.

Reply:
Thinking about it again - what I can do is simply call `handlePendingPodTimeout()` at the end of the `if` and then return.
@xpivarc WDYT? I don't think there's another reason to not return here

Reply:
The flow:
1. handleSignalMigrationAbort get called
2. (pod does not exist) the execution follows and we create pod

Correct me if I see wrong but this is possible flow and I think it would be bad one.

Reply:
Why would it be bad? The target pod would be killed due to timeout as I've written above. Can you elaborate?